packages/uniforms-material/src/SelectField.tsx

```diff
@@ -170,15 +174,20 @@ function Select(props: SelectFieldProps) {
   const Item = native ? 'option' : MenuItem;
   const hasPlaceholder = !!placeholder;
   const hasValue = value !== '' && value !== undefined;
-  const filteredProps = omit(filterDOMProps(props), [
+  const filteredOmittedItems = [
     'checkboxes',
     'disableItem',
     'fullWidth',
     'helperText',
     'margin',
     'textFieldProps',
     'variant',
-  ]);
+  ] as const;
+  const filteredProps = omit(filterDOMProps(props), filteredOmittedItems);
+  const filteredPropsOmitted = filteredProps as Omit<
+    typeof filteredProps,
+    typeof filteredOmittedItems[number]
+  >;
```

Comment:
I've debugged it myself, and it looks like only the `checkboxes` and `disableItem` properties are a problem for the `omit` inference. Here's my approach, that reduces the fuss of type handling:
```diff
   const filteredProps = omit(filterDOMProps(props), [
-    'checkboxes',
-    'disableItem',
+    'checkboxes' as never,
+    'disableItem' as never,
     'fullWidth',
     'helperText',
     'margin',
     'textFieldProps',
     'variant',
   ]);
```

It works because we "hide" these two problematic properties from `omit` and let it do the trick. Let me know if you like it, as I think it's much cleaner. (Of course, it requires a solid comment _why_ it's there.)